tracardi/process_engine/action/v1/notification_plugin_action.py

```diff
@@ -0,0 +1,152 @@
+import json
+from json import JSONDecodeError
+from typing import Optional
+
+import aiohttp
+
+from tracardi.domain.resources.token import Token
+
+from tracardi.service.tracardi_http_client import HttpClient
+
+from tracardi.service.storage.driver import storage
+
+from tracardi.service.plugin.domain.register import Plugin, Spec, Form, FormGroup, FormField, FormComponent, MetaData, \
+    Documentation, PortDoc
+
+from tracardi.domain.named_entity import NamedEntity
+
+from tracardi.domain.resource import ResourceCredentials
+from tracardi.service.plugin.domain.config import PluginConfig
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+
+
+class Config(PluginConfig):
+    source: NamedEntity
+    template_name: str
+    subscriber_id: str
+    recipient_email: Optional[str] = ""
+    payload: Optional[str] = "{}"
+
+
+def validate(config: dict) -> Config:
+    return Config(**config)
+
+
+class NotificationGeneratorAction(ActionRunner):
+
+    @staticmethod
+    async def build(**kwargs) -> 'NotificationGeneratorAction':
+        config = validate(kwargs)
+        resource = await storage.driver.resource.load(config.source.id)
+        return NotificationGeneratorAction(config, resource.credentials)
+
+    def __init__(self, config: Config, credentials: ResourceCredentials):
+        self.config = config
+        self.credentials = credentials.get_credentials(self, output=Token)  # type: Token
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        url = 'https://api.novu.co/v1/events/trigger'
+        dot = self._get_dot_accessor(payload)
+        timeout = aiohttp.ClientTimeout(total=15)
+        params = {
+            "name": self.config.template_name,
+            "to": {
+                "subscriberId": dot[self.config.subscriber_id],
+                "email": dot[self.config.recipient_email]
+            },
+            "payload": json.loads(self.config.payload)
+        }
+        print(params)
+        async with HttpClient(timeout=timeout, retries=self.node.on_connection_error_repeat) as client:
```

Comment:
retries should be passed as a positional argument, **kwargs goes to ClientSession(**kwargs), which is member of HttpClient class

Reply:
OK statuses should also be passed as a positional argument (second one). Use array [200, 201...] from below.